saltproc/process.py

```diff
@@ -72,44 +78,60 @@ def check_mass_conservation(self):
 
         Parameters
         ----------
         inflow : Materialflow obj
-            Target material stream to remove poisons from.
+            Material flowing into the processing system component.
 
         Returns
         -------
-        Materialflow object
+        waste_stream : Materialflow object
             Waste stream from the reprocessing system component.
+        thru_flow : Materialflow object
+            Remaining material flow that will pass through the
+            reprocessing system component.
 
         """
         waste_nucvec = {}
-        out_nucvec = {}
+        thru_nucvec = {}
         # print("Xe concentration in inflow before % f g" % inflow['Xe136'])
         # print("Current time %f" % (t))
-        for iso in inflow.comp.keys():
-            el_name = pyname.serpent(iso).split('-')[0]
-            if el_name in self.efficiency:
-                # Evaluate removal efficiency for el_name (float)
-                self.efficiency[el_name] = self.calc_rem_efficiency(el_name)
-                out_nucvec[iso] = \
-                    float(inflow.comp[iso]) * \
-                    float(1.0 - self.efficiency[el_name])
-                waste_nucvec[iso] = \
-                    float(inflow[iso]) * self.efficiency[el_name]
+
+        for nuc in inflow.comp.keys():
+            nuc_name = pyname.serpent(nuc).split('-')[0]
+            if nuc_name in self.efficiency:
+                # Evaluate removal efficiency for nuc_name (float)
+                self.efficiency[nuc_name] = \
+                    self.calculate_removal_efficiency(nuc_name)
+
+                thru_nucvec[nuc] = \
+                    float(inflow.comp[nuc]) * \
+                    float(1.0 - self.efficiency[nuc_name])
+
+                waste_nucvec[nuc] = \
+                    float(inflow[nuc]) * self.efficiency[nuc_name]
+
+            # Assume zero removal
             else:
-                out_nucvec[iso] = float(inflow.comp[iso])
-                waste_nucvec[iso] = 0.0  # zeroes everywhere else
-        waste = Materialflow(waste_nucvec)
-        inflow.mass = float(inflow.mass - waste.mass)
-        inflow.comp = out_nucvec
-        inflow.norm_comp()
-        print("Xe concentration in inflow after %f g" % inflow['Xe136'])
-        print("Waste mass %f g\n" % waste.mass)
-        del out_nucvec, waste_nucvec, el_name
-        return waste
+                thru_nucvec[nuc] = float(inflow.comp[nuc])
+                waste_nucvec[nuc] = 0.0
+
+        waste_stream = Materialflow(waste_nucvec)
+        # preserve inflow attributes
+        thru_flow = deepcopy(inflow)
+        thru_flow.comp = thru_nucvec
+        thru_flow.density = inflow.density
+        thru_flow.mass = float(inflow.mass - waste_stream.mass)
+        thru_flow.norm_comp()
+
+        print("Xe concentration in thruflow: %f g" % thru_flow['Xe136'])
+        print("Waste mass: %f g\n" % waste_stream.mass)
```

Comment:
This makes me think that the above-commented print statements should be uncommented.